### PR TITLE
vtep:  VTEP map implementation

### PR DESCRIPTION
**Pull request by vincentmli**
Current VTEP integration uses macro hack for VTEP and cause issue in https://github.com/cilium/ebpf/issues/553 when number of VTEP devices `> 2`, not only that, the implementation also populate `ipcache` with VTEP and result in not supporting Cilium network policy, see  https://github.com/cilium/cilium/issues/17694. Let's implement VTEP specific hash map to address above issues.

Please ensure your pull request adheres to the following guidelines:

- [ ] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [ ] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Thanks for contributing!

Fixes: #issue-number

```release-note
vtep: VTEP map implementation to improve VTEP integration feature
```

### CODE REVIEW

**Comment by vincentmli** [comment]
running into strange BPF hash map creation issue with this draft implementation that is identical to tunnel map implementation, there are two hash maps get created `VTEP_MAP` and `cilium_vtep_map`:

 kubectl exec -it cilium-g6zx7 -n kube-system -- bpftool map list --bpffs -j | jq '.[] | select( .type == "hash" )' | jq 

```
{
  "id": 6811,
  "type": "hash",
  "name": "VTEP_MAP",
  "flags": 1,
  "bytes_key": 4,
  "bytes_value": 16,
  "max_entries": 65536,
  "bytes_memlock": 1572864,
  "frozen": 0,
  "btf_id": 510,
  "pinned": [
    "/sys/fs/bpf/tc/globals/VTEP_MAP" <===this is very odd and seems to be from bpf/lib/maps.h
  ]
}
{
  "id": 9956,
  "type": "hash",
  "flags": 1,
  "bytes_key": 20,
  "bytes_value": 20,
  "max_entries": 65536,
  "bytes_memlock": 2621440,
  "frozen": 0,
  "pinned": [
    "/sys/fs/bpf/tc/globals/cilium_tunnel_map" <===tunnel_map does not have TUNNEL_MAP like above
  ]
}
{
  "id": 9957,
  "type": "hash",
  "flags": 1,
  "bytes_key": 4,
  "bytes_value": 16,
  "max_entries": 8,
  "bytes_memlock": 4096,
  "frozen": 0,
  "pinned": [
    "/sys/fs/bpf/tc/globals/cilium_vtep_map". <=====this should be the only one from pkg/maps/vtep/vtep.go
  ]
}

```
and `VTEP_MAP` is empty

```
 kubectl exec -it cilium-g6zx7 -n kube-system -- bpftool map dump id  6811

[]
```
`cilium_vtep_map` has records:

```
kubectl exec -it cilium-g6zx7 -n kube-system -- bpftool map dump id  9957

key: 0a 01 04 00  value: 82 36 4c 98 2e 57 00 00  0a a9 48 eb 00 00 00 00
key: 0a 01 05 00  value: 82 36 4c 98 2e 56 00 00  0a a9 48 ec 00 00 00 00
Found 2 elements

```

I suspect above empty `VTEP_MAP` may cause packet drop problem here due to 

```
                vtep = map_lookup_elem(&VTEP_MAP, &vkey);
                if (!vtep)
                        return DROP_NO_TUNNEL_ENDPOINT;
```

DROP: 98 bytes, reason No tunnel/encapsulation endpoint (datapath BUG!), identity 61928->world

```
Ethernet        {Contents=[..14..] Payload=[..86..] SrcMAC=52:4a:f5:8d:ec:9c DstMAC=2e:4b:59:d2:4b:03 EthernetType=IPv4 Length=0}
IPv4    {Contents=[..20..] Payload=[..64..] Version=4 IHL=5 TOS=0 Length=84 Id=10562 Flags=DF FragOffset=0 TTL=64 Protocol=ICMPv4 Checksum=63239 SrcIP=10.0.0.115 DstIP=10.1.5.236 Options=[] Padding=[]}
ICMPv4  {Contents=[..8..] Payload=[..56..] TypeCode=EchoRequest Checksum=18256 Id=23815 Seq=1}
  Failed to decode layer: No decoder for layer type Payload
CPU 02: MARK 0x0 FROM 1095 DROP: 98 bytes, reason No tunnel/encapsulation endpoint (datapath BUG!), identity 61928->world

```

**Comment by vincentmli** [comment]
and if I add `cDefinesMap["VTEP_MAP"] = vtep.Name` in `pkg/datapath/linux/config/config.go`, both `VTEP_MAP` and `cilium_vtep_map` is empty from `bpftool map dump`, but `cilium map get cilium_vtep_map` shows records.

**Comment by vincentmli** [comment]
> running into strange BPF hash map creation issue with this draft implementation that is identical to tunnel map implementation, there are two hash maps get created `VTEP_MAP` and `cilium_vtep_map`:

I don't see `VTEP_MAP` anymore under `/sys/fs/bpf/tc/globals` anymore with current updated draft code, maybe `VTEP_MAP` is leftover from previous broken draft code.

**Comment by vincentmli** [comment]
rebase this draft PR on master with dynamic ARP and MAC refactor PRs merged

**Comment by vincentmli** [comment]
Hi @joestringer, I think this vtep hash map implementation PR is ready for review :)

**Comment by vincentmli** [comment]
change `VTEP_MASK` to `VtepCIDR.Mask` instead of hard coding `0xffffff`, this would allow user to specify VtepCIDR with any mask they want, for example `/30` instead of hard code `/24` 

```
  enable-vtep:   "true"
  vtep-endpoint: "10.169.72.238 10.169.72.236"
  vtep-cidr:     "10.1.1.0/30   10.1.1.4/30"
  vtep-mac:      "82:36:4c:98:2e:55 82:36:4c:98:2e:56"

```
result VTEP_MASK

```
grep 'VTE_MASK' /var/run/cilium/state/globals/node_config.h 
#define VTEP_MASK 0xfcffffff

```
vtep map 
```
IP PREFIX/ADDRESS   VTEP
10.1.1.4            vtepmac=82:36:4C:98:2E:56 tunnelendpoint=10.169.72.236   
10.1.1.0            vtepmac=82:36:4C:98:2E:55 tunnelendpoint=10.169.72.238   
```

diff 

```
diff --git a/pkg/datapath/linux/config/config.go b/pkg/datapath/linux/config/config.go
index b3faf273a9..bfcab15999 100644
--- a/pkg/datapath/linux/config/config.go
+++ b/pkg/datapath/linux/config/config.go
@@ -544,7 +544,11 @@ func (h *HeaderfileWriter) WriteNodeConfig(w io.Writer, cfg *datapath.LocalNodeC
                cDefinesMap["ENABLE_VTEP"] = "1"
                cDefinesMap["VTEP_MAP"] = vtep.Name
                cDefinesMap["VTEP_MAP_SIZE"] = fmt.Sprintf("%d", vtep.MaxEntries)
-               cDefinesMap["VTEP_MASK"] = fmt.Sprintf("%#x", vtep.VtepMask)
+               if len(option.Config.VtepCIDRs) > 0 {
+                       m := option.Config.VtepCIDRs[0].Mask
+                       cDefinesMap["VTEP_MASK"] = fmt.Sprintf("%#x", byteorder.NetIPv4ToHost32(net.IP(m)))
+               }
+
        }
 
        vlanFilter, err := vlanFilterMacros()
diff --git a/pkg/maps/vtep/vtep.go b/pkg/maps/vtep/vtep.go
index 6cfd693fa2..52cd1b5574 100644
--- a/pkg/maps/vtep/vtep.go
+++ b/pkg/maps/vtep/vtep.go
@@ -23,9 +23,6 @@ const (
 
        // Name is the canonical name for the VTEP map on the filesystem.
        Name = "cilium_vtep_map"
-
-       //VtepMask 0xffffff
-       VtepMask = 16777215
 )
```

File: pkg/option/config.go
Comment (joestringer) [inline pkg/option/config.go]: It might be nice to define this `8` in `pkg/defaults` instead, so then the same definition can be used both here in the `pkg/option` as well as `pkg/maps/vtep`.

File: daemon/cmd/datapath.go
Comment (joestringer) [inline daemon/cmd/datapath.go]: extra empty line not needed here.

File: pkg/maps/vtep/logfields.go
Comment (joestringer) [inline pkg/maps/vtep/logfields.go]: This is good style if the fields are unique to this package, but for each of these fields there is already a canonical definition in `pkg/logging/logfields`. Could you use those instead?

File: pkg/datapath/linux/config/config.go
Comment (joestringer) [inline pkg/datapath/linux/config/config.go]: It seems a bit odd to me to only respect the mask specified in the first VTEP CIDR, and apply those to all of the entries. Should the VTEP CIDR mask be an independent configuration parameter to make it clearer that it applies to all VTEP CIDRs in the configuration?

**Review by joestringer — CHANGES_REQUESTED** [review]
Mostly minor nits, but this looks great 👍 Given that this no longer relies on modifying the ipcache, it should also no longer impact policy handling for this traffic so I think this will also solve #17694 ?

File: pkg/datapath/linux/config/config.go
Comment (vincentmli) [reply pkg/datapath/linux/config/config.go]: I did this to write less and simple code :), you mean I could add an independent configuration parameter  `vtep-mask` like below ? 

```
  enable-vtep:   "true"
  vtep-endpoint: "10.169.72.34  10.169.72.34"
  vtep-cidr:     "10.0.66.0/30  10.1.1.0/30"
  vtep-mac:      "00:50:56:a0:7d:d8 00:50:56:a0:7d:d8"
  vtep-mask:    255.255.255.252
```

File: pkg/option/config.go
Comment (vincentmli) [reply pkg/option/config.go]: done

File: daemon/cmd/datapath.go
Comment (vincentmli) [reply daemon/cmd/datapath.go]: done

File: pkg/maps/vtep/logfields.go
Comment (vincentmli) [reply pkg/maps/vtep/logfields.go]: done

**Comment by vincentmli** [comment]
> #17694

I did some test with this PR to see if resolved issue in https://github.com/cilium/cilium/issues/17694, specifically when `toCIDR` is `>` or `<=` vtepCIDR, I think the `>` case is resolved, I don't remember the `<=` case test result, I will have to test it again

**Comment by vincentmli** [comment]
> I did some test with this PR to see if it resolved issue in #17694, specifically when `toCIDR` is `>` or `<=` vtepCIDR, I think the `>` case is resolved, I don't remember the `<=` case test result, I will have to test it again

I confirmed again that when `toCIDR`  is `>` vtepCIDR, it works

cidr.yaml

```
apiVersion: "cilium.io/v2"
kind: CiliumNetworkPolicy
metadata:
  name: "cidr-rule"
spec:
  endpointSelector:
    matchLabels:
      app: netshoot 
  egress:
  - toCIDR:
    - 10.1.0.0/16
```
after applying the policy

```
kubectl exec -it cilium-wzs2s -n kube-system -- cilium bpf policy get --all
Defaulted container "cilium-agent" out of: cilium-agent, clean-cilium-state (init)


POLICY   DIRECTION   LABELS (source:key[=value])   PORT/PROTO   PROXY PORT   BYTES   PACKETS   
Allow    Ingress     reserved:unknown              ANY          NONE         4060    44        
Allow    Ingress     reserved:host                 ANY          NONE         0       0         
Allow    Egress      cidr:10.1.0.0/16              ANY          NONE         0       0         
                     reserved:world    

```

toCIDR `10.1.0.0/16` contains vtepCIDR `10.1.1.0/24` 

```
  enable-vtep:   "true"
  vtep-endpoint: "10.169.72.34 10.169.72.236"
  vtep-cidr:     "10.1.2.0/24  10.1.1.0/24"
  vtep-mac:      "00:50:56:a0:7d:d8 82:36:4c:98:2e:56"
```     

ping ip in vtepCIDR from netshoot

```
kubectl exec -it netshoot -- ping 10.1.1.236
PING 10.1.1.236 (10.1.1.236) 56(84) bytes of data.
64 bytes from 10.1.1.236: icmp_seq=1 ttl=63 time=1.21 ms
64 bytes from 10.1.1.236: icmp_seq=2 ttl=63 time=1.06 ms
```

I am going to test the toCIDR `<=` vtepCIDR case, I suspect it won't work because toCIDR policy on netshoot pod is enforced before vtep map lookup and before vtep code execution, and it should be expected to not work.

File: pkg/datapath/linux/config/config.go
Comment (vincentmli) [reply pkg/datapath/linux/config/config.go]: I added vtep-mask option and pushed the changes :)

**Comment by vincentmli** [comment]
Hi @joestringer I rebased on master and addressed the review change request

File: cilium/cmd/bpf_vtep.go
Comment (qmonnet) [inline cilium/cmd/bpf_vtep.go]:
```suggestion
// Copyright Authors of Cilium
```

File: cilium/cmd/bpf_vtep_list.go
Comment (qmonnet) [inline cilium/cmd/bpf_vtep_list.go]: same

File: cilium/cmd/bpf_vtep_list.go
Comment (qmonnet) [inline cilium/cmd/bpf_vtep_list.go]:
```suggestion
	ciliumMapGet  = "You may run 'cilium map get cilium_vtep_map'.\n"
```

File: cilium/cmd/bpf_vtep.go
Comment (qmonnet) [inline cilium/cmd/bpf_vtep.go]:
```suggestion
	Short: "Manage the VTEP mappings for IP/CIDR <-> VTEP MAC/IP",
```

File: cilium/cmd/bpf_vtep_list.go
Comment (qmonnet) [inline cilium/cmd/bpf_vtep_list.go]:
```suggestion
	Short:   "List VTEP CIDR and their corresponding VTEP MAC/IP",
```

File: cilium/cmd/bpf_vtep_list.go
Comment (qmonnet) [inline cilium/cmd/bpf_vtep_list.go]: Do you need a specific variable for the second part of the message? It looks like you're not reusing it later?
Also you have a double line break (in format string _and_ in the string in the variable).

File: daemon/cmd/datapath.go
Comment (qmonnet) [inline daemon/cmd/datapath.go:495]: Nit: could remove the blank line below

File: pkg/maps/vtep/doc.go
Comment (qmonnet) [inline pkg/maps/vtep/doc.go]:
```suggestion
// Copyright Authors of Cilium
```

File: pkg/maps/vtep/vtep.go
Comment (qmonnet) [inline pkg/maps/vtep/vtep.go]: same

File: daemon/cmd/datapath.go
Comment (qmonnet) [inline daemon/cmd/datapath.go:495]: Why VTEP and no longer IPCache? In what regard is this an improvement? This info should ideally be documented in the commit description, so that we can understand the reason for this change when looking at the history.

File: Documentation/cmdref/cilium-agent.md
Comment (qmonnet) [inline Documentation/cmdref/cilium-agent.md:259]: New options should be documented in `Documentation/operations/upgrade.rst`, can you please add it?

File: bpf/lib/maps.h
Comment (qmonnet) [inline bpf/lib/maps.h]:
```suggestion
```

File: bpf/lib/maps.h
Comment (qmonnet) [inline bpf/lib/maps.h]:
```suggestion
```

**Review by qmonnet — CHANGES_REQUESTED** [review]
Thanks a lot for the contribution! The code looks clean.

Please find a few comments below. In particular, I would suggest adding some context and motivation to the commit log for the first commit, so that people can understand what's going on in the future when looking at the history.

File: cilium/cmd/bpf_vtep.go
Comment (vincentmli) [reply cilium/cmd/bpf_vtep.go]: done

File: daemon/cmd/datapath.go
Comment (vincentmli) [reply daemon/cmd/datapath.go:495]: previous implementation bypass `pkg/ipcache` and update IPCache directly which may cause inconsistency in IPCache, I will update the commit message.

File: Documentation/cmdref/cilium-agent.md
Comment (vincentmli) [reply Documentation/cmdref/cilium-agent.md:259]: the first implementation is only released in pre-release candidate v1.12.0-rc0, , I am hoping after this PR is merged, there will be a final complete VTEP integration release in v1.12.0, updating upgrade.rst seems not needed, I  see `1.11 Upgrade Notes` and `1.12 Upgrade Notes` in `upgrade.rst`, the VTEP integration is not in `1.12` yet.

File: cilium/cmd/bpf_vtep_list.go
Comment (vincentmli) [reply cilium/cmd/bpf_vtep_list.go]: I added the `ciliumMapGet` while debugging VTEP map implementation issue, i think it is not needed now, I can remove it.

**Review by qmonnet — APPROVED** [review]
Thanks a lot!

File: Documentation/cmdref/cilium-agent.md
Comment (joestringer) [reply Documentation/cmdref/cilium-agent.md:259]: Yeah given that it's a new feature in  1.12, no upgrade notes should be necessary.

File: pkg/maps/vtep/vtep.go
Comment (joestringer) [inline pkg/maps/vtep/vtep.go]: Any time that you include another error in an error that is passed up, it should use the wrap formatter (`%w`) so that the callers can properly unwrap it later to identify the underlying error.

```suggestion
		return fmt.Errorf("invalid VTEP MAC: %w", err)
```

File: pkg/maps/vtep/vtep.go
Comment (joestringer) [inline pkg/maps/vtep/vtep.go]: Given that this function is checking for errors, and potentially returning errors if there are badly formatted values, if `ip4 == nil` here then it should return an error something like `fmt.Errorf("Cannot configure VTEP endpoint for IP %v", ip4")`.

File: pkg/maps/vtep/vtep.go
Comment (joestringer) [inline pkg/maps/vtep/vtep.go]: This can be simplified:

```suggestion
	return k.IP.String()
```

File: pkg/maps/vtep/vtep.go
Comment (joestringer) [inline pkg/maps/vtep/vtep.go]: The datapath takes a packet destination, masks it with the VTEP CIDR mask, then performs the lookup. However, this key creation & insert from the control plane directly takes the user input and passes it into the map. I believe this means that if the user specifies `--vtep-cidr=192.168.0.1 --vtep-mask=255.255.255.0` then the lookup will fail, because the map is populated with `192.168.0.1` instead of `192.168.0.0` (ie the masked version). This is a minor usability nit but could catch someone out, I think. Maybe this `NewKey()` function should also apply the mask against the CIDRs when creating the keys, to ensure that they are properly formatted before inserting into the datapath map?

File: pkg/maps/vtep/vtep.go
Comment (joestringer) [inline pkg/maps/vtep/vtep.go]: nit, there's still extra random spaces around function start/end which is unusual for Cilium code.
```suggestion

```

File: pkg/maps/vtep/vtep.go
Comment (joestringer) [inline pkg/maps/vtep/vtep.go]:
```suggestion
```

**Review by joestringer — CHANGES_REQUESTED** [review]
Just a few more minor bits of feedback below to make this code more consistent with the error handling, user input handling, and style of existing code.

File: pkg/maps/vtep/vtep.go
Comment (vincentmli) [reply pkg/maps/vtep/vtep.go]: done

File: pkg/maps/vtep/vtep.go
Comment (vincentmli) [reply pkg/maps/vtep/vtep.go]: since the `newTunnelEndpoint` passed on to `UpdateVTEPMapping` is already validated 
 to be valid IPv4 address in `pkg/option/config.go`, I am wondering if it is necessary to check `newTunnelEndpoint.To4()` error return here, thought?

File: pkg/maps/vtep/vtep.go
Comment (vincentmli) [reply pkg/maps/vtep/vtep.go]: good catch!

Comment (vincentmli) [reply pkg/maps/vtep/vtep.go]: sorry be to rough on golang here :), I did this `ip4 = ip4 & option.Config.VtepCidrMask` and I got error

```
invalid operation: operator & not defined on ip4 (variable of type net.IP)
```
tried google search proper golang bitwise mask operation on IP, no result

Comment (joestringer) [reply pkg/maps/vtep/vtep.go]: I would suggest using the proper IPMask type to apply the masking: https://pkg.go.dev/net#IP.Mask

File: pkg/maps/vtep/vtep.go
Comment (joestringer) [reply pkg/maps/vtep/vtep.go]: Fair point, it shouldn't be necessary if the inputs are properly validated earlier. It's just odd to see that this function returns an error, but then there's an error case here that is not handled.

Though on closer inspection, it looks like this is the validation that occurs to check the addressing family in the option validation:

```
		if strings.Contains(ep, ":") {
			return fmt.Errorf("VTEP integration IPv6 not supported: %v", ep)
		}
```

I think that the option validation should do the same operation as you're doing here - after the `net.ParseIP()` call, it should attempt `endpoint.To4()` and check that it is non-nil. That way we can be sure that this code will always return a non-nil ipv4 address.

This way, you shouldn't need this custom check for the colon, but you can rely on the standard library checks for IP parsing & converting into IPv4 addresses.

File: pkg/maps/vtep/vtep.go
Comment (vincentmli) [reply pkg/maps/vtep/vtep.go]: silly me :) there is already Mask function 
```
      if ip4 := ip.To4(); ip4 != nil {
+               ip4.Mask(net.IPMask(option.Config.VtepCidrMask))
                copy(result.IP[:], ip4)
        }
```

**Comment by vincentmli** [comment]
> rebase on master and pushed new changes according to most recent code review, it is weird having problem running the new build local cilium agent, same code having no problem on not updated master

had to rebuild my local cilium operator image `make docker-operator-image` and update the command to 

```
        command:
        - cilium-operator
```

**Review by joestringer — APPROVED** [review]
LGTM

**Comment by vincentmli** [comment]
@joestringer @qmonnet thanks for approving the PR, could either one of you trigger the test? i am hoping this PR will be merged in master and eventually catch the v1.12.0 final release :)

**Comment by joestringer** [comment]
gke-stable hit #13917, known occasional flake.

Reviews are in for most codeowners, practically I doubt that there is much more feedback likely to come in from the remaining codeowners given how simple the changes are in those areas of the code.

I'm fine to merge this as-is.

**Comment by joestringer** [comment]
Actually @vincentmli it looks like there are conflicts on this vs master. Can you rebase?

**Comment by vincentmli** [comment]
> > Actually @vincentmli it looks like there are conflicts on this vs master. Can you rebase?
> 
> it looks `Documentation/cmdref/cilium_bpf.md` needs to be updated, I am rebasing on master and running `make -C Documentation update-cmdref` to update cilium_bpf.md, I have noticed it before, want to mention here that `make -C Documentation update-cmdref` seems triggering not only update `Documentation/cmdref/`, but also actually compile/build the cilium/bpf directory, is that intended behavior? I thought it should just only update the `Documentation/cmdref/`, or maybe I should just run `cilium cmdref` that I just saw from the newly created `Documentation/cmdref/cilium_bpf.md`, now it should be updated, thanks @joestringer


run into problem https://app.netlify.com/sites/cilium-doc/deploys/6255c491486e4b0008e38437

```
11:28:12 AM: $ make html-netlify
11:28:12 AM: cp -r ../api _api
11:28:12 AM: SKIP_LINT=1 ./check-build.sh html
11:28:12 AM: Skipping syntax and spelling validations...
11:28:12 AM: Building documentation (html)...
11:28:59 AM: /opt/build/repo/Documentation/cmdref/cilium.md:22: WARNING: 'myst' reference target not found: cilium_bpf.md
11:28:59 AM: /opt/build/repo/Documentation/cmdref/cilium_bpf_bandwidth.md:23: WARNING: 'myst' reference target not found: cilium_bpf.md
11:28:59 AM: /opt/build/repo/Documentation/cmdref/cilium_bpf_ct.md:23: WARNING: 'myst' reference target not found: cilium_bpf.md
11:28:59 AM: /opt/build/repo/Documentation/cmdref/cilium_bpf_egress.md:23: WARNING: 'myst' reference target not found: cilium_bpf.md
```

I tried `make html-netlify` in local repo, but `make: *** No rule to make target 'html-netlify'.  Stop.`

**Comment by joestringer** [comment]
> `make -C Documentation update-cmdref` seems triggering not only update `Documentation/cmdref/`, but also actually compile/build the cilium/bpf directory, is that intended behavior?

update-cmdref needs to run the cilium binary to generate the new command reference, so my guess is that it is directly relying on the main `cilium-agent` (and `cilium/` in-pod CLI and other binaries), compiling those, and hence implicitly also depending on cilium/bpf. It's not strictly required, but I can sorta see why this may be happening.

**Comment by joestringer** [comment]
I suspect that the netlify breakage may be introduced by #19348 (cc @qmonnet ): https://app.netlify.com/sites/cilium-doc/deploys/6255c491486e4b0008e38437 

Locally I actually get some other error, not sure quite what's going on there (might just be that I haven't pip installed the right dependencies locally though?)
```
$ make -C Documentation/ html-netlify
Skipping syntax and spelling validations...
Building documentation (html)...

Configuration error:
There is a programmable error in your configuration file:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/sphinx/config.py", line 328, in eval_config_file
    exec(code, namespace)
  File "/home/joe/git/cilium/Documentation/conf.py", line 23, in <module>
    import semver
ModuleNotFoundError: No module named 'semver'

make: *** [Makefile:97: html-netlify] Error 2
```

File: Documentation/cmdref/cilium_bpf.md
Comment (joestringer) [inline Documentation/cmdref/cilium_bpf.md]: Wait, no, this is probably why the netlify build fails (sorry @qmonnet for the noise!)

Why is this file being deleted in this PR?

Comment (vincentmli) [reply Documentation/cmdref/cilium_bpf.md]: ah, to resolve the conflicts, I rebased on master and `git rm Documentation/cmdref/cilium_bpf.md`, then `make -C Documentation update-cmdref`, after that, when I saw  `Documentation/cmdref/cilium_bpf.md` get created, I pushed the PR, I should have `git add Documentation/cmdref/cilium_bpf.md; git commit; git push...;" like I usually do

Comment (vincentmli) [reply Documentation/cmdref/cilium_bpf.md]: it looks I also need to git add Documentation/cmdref/cilium_bpf_vtep.md, Documentation/cmdref/cilium_bpf_vtep_list.md that are created by `make -C Documentation update-cmdref`, I don't remember if  I did this previously for this PR, but now it seems to be the correct way to do it.

**Review by joestringer — APPROVED** [review]
Ah yes, I didn't realize that the vtep command reference was missing before. The latest changes look good, thanks!

**Comment by joestringer** [comment]
Netlify and the docs target are both happy now, and the only changes since my last review were changes to the command reference in the docs. This is now good to merge. Thanks @vincentmli !

**Pull request merged**
